src/context/apiClient/apiClientContextController/interceptors/responseInterceptors.ts

```diff
@@ -1,5 +1,60 @@
-import { AxiosResponse } from 'axios';
+import axios, { AxiosError, AxiosResponse } from 'axios';
+import decode from 'jwt-decode';
+
+import { RefreshMutationResponse } from 'api/actions/auth/authActions.types';
+import { authStorage } from 'context/auth/authStorage/AuthStorage';
+import { APIErrorOutput, ExtendedAxiosRequestConfig } from 'api/types/types';
+import { narrowErrorData } from 'api/axios/narrowErrorData';
 
 export const responseSuccessInterceptor = (response: AxiosResponse) => response;
 
-export const responseFailureInterceptor = async (error: any) => Promise.reject(error); // eslint-disable-line @typescript-eslint/no-explicit-any
+export const responseFailureInterceptor = async (error: AxiosError): Promise<APIErrorOutput> => {
```

Comment:
1 little thing, can we add a possibility to define different strategies here, with narrowErrorData as the default value?